### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by memoryleak47**
This mirrors the API cleanups from here https://github.com/memoryleak47/minirust-tooling/commit/d2a4cec394c569c26392eb8bf422938b7ef4a0e3.

### CODE REVIEW

File: lang/values.md [hunk 1]
```diff
@@ -308,16 +308,16 @@ impl<Provenance> DefinedRelation for AbstractByte<Provenance> {
     fn le_defined(self, other: Self) -> bool {
         use AbstractByte::*;
         match (self, other) {
-            /// `Uninit <= _`: initializing something makes it "more defined".
```
Comment (memoryleak47) [inline lang/values.md:311]: match-arms don't generate doc-comments.
This caused a warning.

File: prelude/main.md [hunk 2]
```diff
@@ -17,32 +19,21 @@ pub enum TerminationInfo {
 /// `Option`/`Result`-returning function.
 macro_rules! throw {
     ($($tt:tt)*) => {
-        specr::yeet!(())
```
Comment (memoryleak47) [inline prelude/main.md:20]: `specr::yeet` only called `do yeet` anyways.

File: prelude/main.md [hunk 3]
```diff
@@ -17,32 +19,21 @@ pub enum TerminationInfo {
...
-/// The enum `Signedness` expresses whether an integer value is signed or unsigned.
-pub use specr::Signedness;
-pub use Signedness::{Signed, Unsigned};
-
-/// Whether a pointer/reference/allocation is mutable or immutable.
-pub enum Mutability {
-    Mutable,
-    Immutable,
-}
-pub use Mutability::*;
```
Comment (memoryleak47) [inline prelude/main.md:47]: Do you want this to stay here?
I'd prefer moving this into the `libspecr::prelude`, along with `Endianness` and the `Signedness` definition.

File: lang/well-formed.md [hunk 4]
```diff
@@ -75,7 +75,7 @@ impl Type {
             }
             Array { elem, count } => {
                 elem.check_wf::<M>()?;
-                elem.size::<M>() * count;
+                ensure(M::valid_size(elem.size::<M>() * count))?;
```
Comment (RalfJung) [inline lang/well-formed.md:78]: We have already done `ensure(M::valid_size(size))?` above, isn't that enough?

**Pull request merged**